Remove unnecessary i18nDir from hugo.toml

SayakMukhopadhyay opened the pull request:
### Description
This PR removes the unnecessary `i18nDir` property from `hugo.toml`.

### Issue
Closes: #49649

sftim commented:
Hmm, https://deploy-preview-49643--kubernetes-io-main-staging.netlify.app/bn/releases/ doesn't look happy does it? (compare https://kubernetes.io/bn/releases/)

SayakMukhopadhyay commented:
Nope, it seems like its not picking up the website defined localisation keys

sftim commented on i18n/bn/bn.toml:
Empty file?

SayakMukhopadhyay replied:
oops, something went wrong between all the copies and deletes. Should be fixed now

SayakMukhopadhyay commented:
This time, it seems to work. I removed all the symlinks and moved all the files from `data/i18n` to `i18n`. I still think overrides aren't working so let me test that first.

SayakMukhopadhyay commented:
This PR is ready for review. It contains 55 file changes but most of them (52 files) are just the translations moving from `data/i18n` to `i18n`.

The other 3 files include 2 files for removing the `OWNERS` from `data/i18n` and adding it back to `i18n` (git doesn't consider it as a move) and the 3rd file is the update to the `hugo.toml` which adds hugo mounts to it.

sftim reviewed (commented):
How confident are you that README and contributor docs changes are included here, @SayakMukhopadhyay ?

SayakMukhopadhyay commented:
> How confident are you that README and contributor docs changes are included here, @SayakMukhopadhyay ?

@sftim They aren't. I was thinking about updating the README about the changes in here in a new PR, unless you would like to have them in this PR itself.

sftim commented on hugo.toml:
```diff
@@ -517,3 +516,74 @@ languagedirection = "ltr"
 languageNameLatinScript = "Tiếng Việt"
 description = "Giải pháp điều phối container trong môi trường production"
 
```
We should add a comment here, I think.
What happens if someone creates, eg `i18n/de/en.toml`?

The comment I have in mind warns people not to try that.

SayakMukhopadhyay replied:
Sounds good! Done!

sftim reviewed (commented):
Maybe the individual localization directories should have _filters_ that only match TOML files.

See https://www.kubernetes.dev/docs/guide/owners/#filters

SayakMukhopadhyay commented:
> Maybe the individual localization directories should have _filters_ that only match TOML files.
> 
> See https://www.kubernetes.dev/docs/guide/owners/#filters

Is there any possibility of the individual localisation directories containing anything other than the <language>.toml files? I think any other file should be discouraged right? In which case, should we add a filter for OWNERS?

sftim commented:
> Is there any possibility of the individual localisation directories containing anything other than the .toml files? I think any other file should be discouraged right? In which case, should we add a filter for OWNERS?

A team could make a mistake, or might update their `OWNERS` file (which we don't want).

SayakMukhopadhyay commented:
> > Is there any possibility of the individual localisation directories containing anything other than the .toml files? I think any other file should be discouraged right? In which case, should we add a filter for OWNERS?
> 
> A team could make a mistake, or might update their `OWNERS` file (which we don't want).

Agreed, I think we should do it in a fresh PR. There are a few changes in those `OWNERS` files (like https://github.com/kubernetes/website/blob/5de96d97b746cf156efd59ea343be08ff328048e/data/i18n/pl/OWNERS is missing the `area/localization` label).

SayakMukhopadhyay commented:
> How confident are you that README and contributor docs changes are included here, @SayakMukhopadhyay ?

@sftim I have also added a commit in this PR to update the docs at https://kubernetes.io/docs/contribute/localization/ to reflect the changes. I think that's all there is regarding documentation around localisation. I will squash the commits into 2 commits, 1 for the changes and the other for the doc updates.

sftim commented:
/retitle Remove unnecessary i18nDir from hugo.toml

(Markdown in commit messages doesn't really work)

sftim commented:
/lgtm

Localization team, any tech lead, or a chair acting on behalf of tech leads can unhold this at will.

divya-mohan0209 commented:
@SayakMukhopadhyay Thanks for this PR. Could we address the merge conflicts?

SayakMukhopadhyay commented:
@divya-mohan0209 rebased. The lgtm label has been removed due to it though.

natalisucks commented:
@SayakMukhopadhyay np, we got you
/lgtm

sftim commented:
/approve

divya-mohan0209 commented:
@SayakMukhopadhyay I still see merge conflicts on here for the en.toml file. Could you address those? One done, I'd be happy to take Tim and Natali's approvals and merge it immediately.

SayakMukhopadhyay commented:
@divya-mohan0209 I have made another rebase but any PRs that has changes to the translations strings is causing potential merge conflicts. It's kind of odd though as I don't get any merge conflicts while rebasing.

But yeah, if other PRs with changes to the strings get merged, this PR will become problematic once again.

divya-mohan0209 commented:
@SayakMukhopadhyay : Thanks for this! In the interest of getting this PR merged before any other localisation ones are merged, I'll take the approve & lgtm from previous comments.
/lgtm
/approve

SayakMukhopadhyay commented:
/hold cancel
